### PR TITLE
Type validation support

### PR DESCRIPTION
**Pull request by SimonCockx**

### CODE REVIEW

File: rosetta-ide/src/main/java/com/regnosys/rosetta/ide/highlight/RosettaHighlightingCalculator.xtend
Comment (SimonCockx) [inline rosetta-ide/src/main/java/com/regnosys/rosetta/ide/highlight/RosettaHighlightingCalculator.xtend:1]: Deprecated - only used by Eclipse

File: rosetta-ide/src/main/java/com/regnosys/rosetta/ide/highlight/RosettaHighlightingStyles.java
Comment (SimonCockx) [inline rosetta-ide/src/main/java/com/regnosys/rosetta/ide/highlight/RosettaHighlightingStyles.java:1]: Deprecated - only used by Eclipse

File: rosetta-lang/model/Rosetta.xcore [hunk 1]
```diff
@@ -1,5 +1,5 @@
 @Ecore(nsURI="http://www.rosetta-model.com/Rosetta")
-@GenModel(fileExtensions="rosetta", modelDirectory="/rosetta-lang/emf-gen/main/java", operationReflection="false",
+@GenModel(fileExtensions="rosetta", modelDirectory="/com.regnosys.rosetta/emf-gen/main/java", operationReflection="false",
```
Comment (SimonCockx) [inline rosetta-lang/model/Rosetta.xcore:2]: Fix for auto code generation while developing in Eclipse

File: rosetta-lang/model/RosettaExpression.xcore [hunk 2]
```diff
@@ -1,10 +1,12 @@
 @Ecore(nsURI="http://www.rosetta-model.com/RosettaExpression")
-@GenModel(fileExtensions="rosetta", modelDirectory="/rosetta-lang/emf-gen/main/java", operationReflection="false",
+@GenModel(fileExtensions="rosetta", modelDirectory="/com.regnosys.rosetta/emf-gen/main/java", operationReflection="false",
```
Comment (SimonCockx) [inline rosetta-lang/model/RosettaExpression.xcore:2]: Fix for auto code generation while developing in Eclipse

File: rosetta-lang/model/RosettaSimple.xcore [hunk 3]
```diff
@@ -1,5 +1,5 @@
 @Ecore(nsURI="http://www.rosetta-model.com/RosettaSimple")
-@GenModel(fileExtensions="rosetta", modelDirectory="/rosetta-lang/emf-gen/main/java", operationReflection="false",
+@GenModel(fileExtensions="rosetta", modelDirectory="/com.regnosys.rosetta/emf-gen/main/java", operationReflection="false",
```
Comment (SimonCockx) [inline rosetta-lang/model/RosettaSimple.xcore:2]: Fix for auto code generation while developing in Eclipse

File: rosetta-lang/model/Rosetta.xcore [hunk 4]
```diff
@@ -36,10 +36,10 @@ interface RosettaNamed {
 }
 
 interface RosettaTyped {
-	refers RosettaType ^type
+	contains TypeCall ^typeCall
 		
 	derived boolean isTypeInferred get {
-		return ^type === null
+		return ^typeCall === null
```
Comment (SimonCockx) [inline rosetta-lang/model/Rosetta.xcore:42]: Don't need hats anymore

File: rosetta-lang/model/RosettaExpression.xcore [hunk 5]
```diff
@@ -34,36 +36,65 @@ interface HasGeneratedInput {
...
+	
+	op Object containedValue() {
+		return value
+	}
 	op String stringValue() {
+		return Integer.toString(value)
+	}
+}
+
+type Pattern wraps Pattern
```
Comment (SimonCockx) [inline rosetta-lang/model/RosettaExpression.xcore:88]: Add comment: not support.

File: rosetta-lang/src/main/java/com/regnosys/rosetta/RosettaExtensions.xtend [hunk 6]
```diff
@@ -79,7 +91,7 @@ class RosettaExtensions {
 			atts.addAll(data.superType.allNonOverridesAttributes
 				.filter[superAttr| !atts.exists[extendedAttr|					
 					superAttr.name == extendedAttr.name && 
-					superAttr.type == extendedAttr.type && 
+					superAttr.typeCall.type == extendedAttr.typeCall.type &&
```
Comment (SimonCockx) [inline rosetta-lang/src/main/java/com/regnosys/rosetta/RosettaExtensions.xtend:94]: Replace with equality between typeCalls

**Review by hugohills-regnosys — APPROVED** [review]

**Pull request merged**